sources/packages/backend/apps/queue-consumers/src/processors/schedulers/sfas-integration/_tests_/sims-to-sfas-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,249 @@
+      const processingResult = await processor.generateSFASBridgeFile(
+        mockedJob.job,
+      );
+
+      // Assert
+      // Assert process result.
+      expect(processingResult).toEqual([
+        "Process finalized with success.",
+        "Student records sent: 1.",
+        `Uploaded file name: ${expectedFileName}.`,
+      ]);
+      expect(
+        mockedJob.containLogMessages([
+          `Processing data since ${latestBridgeFileDate} until ${mockedCurrentDate}.`,
+          "Found 1 student(s) with updates.",
+          `SIMS to SFAS file ${expectedFileName} has been uploaded successfully.`,
+          `SIMS to SFAS file log has been created with file name ${expectedFileName} and reference date ${mockedCurrentDate}.`,
+        ]),
+      ).toBe(true);
+      const uploadedFile = getUploadedFile(sftpClientMock);
+      const [header, studentRecord, footer] = uploadedFile.fileLines;
+      expect(header).toBe(buildHeader(mockedCurrentDate));
+      expect(footer).toBe("999000000001");
+      expect(studentRecord).toBe(buildStudentRecord(student));
+      // Check the database for creation of SFAS bridge log.
+      const uploadedFileLog = await db.sfasBridgeLog.findOneBy({
+        generatedFileName: expectedFileName,
+        referenceDate: mockedCurrentDate,
+      });
+      expect(uploadedFileLog.id).toBeGreaterThan(0);
+    },
+  );
+
+  it(
+    "Should not generate a SIMS to SFAS bridge file when there is an update on student data but the student " +
+      " does not have any submitted application.",
+    async () => {
+      // Arrange
+      // Create bridge file log.
+      await db.sfasBridgeLog.insert({
+        referenceDate: latestBridgeFileDate,
+        generatedFileName: "DummyFileName.TXT",
+      });
+
+      // Student created with expected first name, last name and more importantly the updated date
+      // to fall between the most recent bridge file date and the mocked current date.
+      await createStudentWithExpectedData(simsDataUpdatedDate);
+
+      // Queued job.
+      const mockedJob = mockBullJob<void>();
+
+      // Mock the current date.
+      MockDate.set(mockedCurrentDate);
+
+      // Act
+      const processingResult = await processor.generateSFASBridgeFile(
+        mockedJob.job,
+      );
+
+      // Assert
+      // Assert process result.
+      expect(processingResult).toEqual([
+        "Process finalized with success.",
+        "Student records sent: 0.",
+        "Uploaded file name: none.",
+      ]);
+      expect(
+        mockedJob.containLogMessages([
+          "There is no SIMS to SFAS updates to process.",
+        ]),
+      ).toBe(true);
+    },
+  );
+
+  /**
+   * Creates a student with expected first name, last name and updated date.
+   * @param expectedUpdatedDate expected updated date.
+   * @param options optional params.
+   * - `expectedCASDetails` expected CAS details.
+   * @returns created student.
+   */
+  async function createStudentWithExpectedData(
```

Comment:
I am not sure if I am following the purpose of this method.
Can we adapt the `saveFakeStudent` to also save the CAS supplier that is required in the same way as CAS?
I am not following also why we need to control the `firstName` and `lastName`.